# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.cpp
# Commit: f82a44c95497
# Full Hash: f82a44c95497c4821a758ba031078945a188bc21
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-06-16 15:49:59
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Refactor wasm::Frame to reduce number of gnarly casts. r=lth,wingo
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79072
# ==============================================================================

diff -r af6b36a497e8 -r f82a44c95497 js/src/wasm/WasmTypes.cpp
--- a/js/src/wasm/WasmTypes.cpp	Tue Jun 16 08:54:41 2020 +0000
+++ b/js/src/wasm/WasmTypes.cpp	Tue Jun 16 09:26:20 2020 +0000
@@ -634,7 +634,7 @@
 
 /* static */
 DebugFrame* DebugFrame::from(Frame* fp) {
-  MOZ_ASSERT(fp->tls->instance->code().metadata().debugEnabled);
+  MOZ_ASSERT(fp->instance()->code().metadata().debugEnabled);
   auto* df =
       reinterpret_cast<DebugFrame*>((uint8_t*)fp - DebugFrame::offsetOfFrame());
   MOZ_ASSERT(fp->instance() == df->instance());